erts/emulator/beam/atom.c

```diff
@@ -233,10 +233,10 @@ static void latin1_to_utf8(byte* conv_buf, const byte** srcp, int* lenp)
 }
 
 /*
- * erts_atom_put() may fail. If it fails THE_NON_VALUE is returned!
+ * erts_atom_put_index() may fail. Returns -1 for bad encoding, -2 for system limit.
  */
-Eterm
-erts_atom_put(const byte *name, int len, ErtsAtomEncoding enc, int trunc)
+int
+erts_atom_put_index(const byte *name, int len, ErtsAtomEncoding enc, int trunc)
```

Comment:
The convention for the `erts_` prefix is to use it for global functions. Older function, or some very often used functions, e.g. `size_object` does not use it though.